gpu/gpu.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM
+# Author: Kalpana Shetty <kalshett@in.ibm.com>
+
+import os
+import shutil
+from avocado import Test, skipUnless
+from avocado.utils import build, process, distro, git
+from avocado.utils.software_manager import SoftwareManager
+
+IS_POWER_NV = 'PowerNV' in open('/proc/cpuinfo', 'r').read()
+
+class GPUTests(Test):
+    """
+    GPU test cases make use of sample test code provided by the
+    CUDA toolkil and other generic test suites cover functional tests.
+    Tester to download and setup CUDA stack which will take care of
+    installing nvidia driver too. 
+    """
+    @skipUnless(IS_POWER_NV,
+                "GPU tests are supported only on PowerNV(POWER9)")
+
+    def process_output(self, output):
+       failed_tests = [] 
+       for line in output.decode("utf-8").splitlines():
+           if "fail" in line:
+               failed_tests.append(line)
+       if failed_tests:
+          self.fail("%s" % failed_tests)
+
+    def run_test(self, run_cmd):
+        os.chdir(self.teststmpdir)
+        output = process.system_output(run_cmd, ignore_status=True)
+        self.process_output(output)
+
+    def setUp(self):
+        """
+        Install pre-requisite packages, compile samples and xgemm test code 
+        """
+        smg = SoftwareManager()
+        self.dist = distro.detect()
+        if process.system('lspci| grep NVIDIA', shell=True, 
+                          ignore_status=True):
```

Comment:
Instead of the shell command with '|' request to use the python code.
run lspci command
parse the output to determine whether NVDIA is there or not.

Reply:
Done